pkg/sql/schemachanger/scpb/elements.proto

```diff
@@ -529,3 +530,65 @@ message IndexData {
 message TablePartitioning {
   uint32 table_id = 1 [(gogoproto.customname) = "TableID", (gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/sql/sem/catid.DescID"];
 }
+
+message Function {
+  message Parameter {
+    option (gogoproto.equal) = true;
+    cockroach.sql.catalog.catpb.FunctionParamClass class = 1 [(gogoproto.nullable) = false];
+    string name = 2;
+    sql.sem.types.T type = 3;
+    string default_expr = 4;
```

Comment:
We have an `Expression` and a `TypeT` message exactly for the purposes of dealing with the references held by a scalar expression and a type. See elsewhere for how we use them.